python-checks/src/main/java/org/sonar/python/checks/cdk/UnrestrictedAdministrationCheck.java

```diff
@@ -0,0 +1,225 @@
+  }
+
+  // Class to handle Dictionary elements as a Map of ResolvedKeyValuePair, useful in case we need to refer to several elements and not only looking for a specific one.
+  // The keys are all resolved String in CdkUtils.ResolvedKeyValuePair.key
+  static class DictionaryAsMap implements CompareInterval {
+    Map<String, CdkUtils.ResolvedKeyValuePair> map = new HashMap<>();
+
+    public static DictionaryAsMap build(SubscriptionContext ctx, DictionaryLiteral dictionary) {
+      DictionaryAsMap dict = new DictionaryAsMap();
+
+      List<CdkUtils.ResolvedKeyValuePair> pairs = dictionary.elements().stream()
+        .map(e -> CdkUtils.getKeyValuePair(ctx, e))
+        .filter(Optional::isPresent).map(Optional::get).collect(Collectors.toList());
+
+      for (CdkUtils.ResolvedKeyValuePair pair : pairs) {
+        pair.key.getExpression(isStringLiteral()).map(StringLiteral.class::cast)
+          .ifPresent(key -> dict.map.put(key.trimmedQuotesValue(), pair));
+      }
+      return dict;
+    }
+
+    public boolean is(String key, Predicate<Expression> valuePredicate) {
+      return map.containsKey(key) && map.get(key).value.hasExpression(valuePredicate);
+    }
+
+    public Optional<Expression> get(String key, Predicate<Expression> valuePredicate) {
+      if (!map.containsKey(key)) {
+        return Optional.empty();
+      }
+      return map.get(key).value.getExpression(valuePredicate);
+    }
+
+    public void addIssue(String key, String message) {
+      if (map.containsKey(key)) {
+        map.get(key).value.addIssue(message);
+      }
+    }
+
+    public Optional<Long> asLong(String name) {
+      return get(name, isNumericLiteral())
+        .map(NumericLiteral.class::cast)
+        .map(NumericLiteral::valueAsLong);
+    }
+  }
+
+  static class Call implements CompareInterval {
+    SubscriptionContext ctx;
+    CallExpression callExpression;
+
+    public Call(SubscriptionContext ctx, CallExpression callExpression) {
+      this.ctx = ctx;
+      this.callExpression = callExpression;
+    }
+
+    boolean hasArgument(String name, Predicate<Expression> predicate) {
+      return getArgument(ctx, callExpression, name).filter(flow -> flow.hasExpression(predicate)).isPresent();
+    }
+
+    public Optional<Long> asLong(String name) {
+      return getArgument(ctx, callExpression, name)
+        .flatMap(flow -> flow.getExpression(isNumericLiteral()))
+        .map(NumericLiteral.class::cast)
+        .map(NumericLiteral::valueAsLong);
+    }
+  }
+
+  // Define the inInterval method used by both Call and DictionaryAsMap classes
+  interface CompareInterval {
+    Optional<Long> asLong(String name);
+
+    default boolean inInterval(String minName, String maxName, long[] numbers) {
+      Optional<Long> min = asLong(minName);
+      Optional<Long> max = asLong(maxName);
+
+      if (min.isEmpty() || max.isEmpty()) {
+        return false;
+      }
+
+      return isInInterval(min.get(), max.get(), numbers);
+    }
```

Comment:
I'm not sure if it's a nice way to have such a generic method inside an interface like this.